### PR TITLE
Bug 1932105: pkg/operator/sync: More logging for syncRequiredMachineConfigPools

### PR DESCRIPTION
**Pull request by wking**
To help with debugging [rhbz#1932105][1], which we're still seeing in CI.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1932105

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
let's merge this to do some testing
/hold cancel

**Review by kikisdeliveryservice — APPROVED** [review]
adding the extra logging to help confirm our theory that checking version.hash isn't sufficient for upgrades.

/lgtm
/hold cancel

File: pkg/operator/status.go [hunk 1]
```diff
@@ -391,6 +391,11 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
 		if ok && v != version {
 			return fmt.Errorf("controller version mismatch for %s expected %s has %s: %v", mcName, version, v, machineConfigPoolStatus(pool))
 		}
+		if ok {
+			glog.Infof("pool %s references machine config %s at the expected %s", pool.GetName(), mcName, version)
+		} else {
+			glog.Infof("pool %s references machine config %s, which does not declare a version", pool.GetName())
```
Comment (kikisdeliveryservice) [inline pkg/operator/status.go:397]: error here missing the mcName:
``` pkg/operator/status.go:397:4: printf: Infof format %s reads arg #2, but call has 1 arg (govet)
			glog.Infof("pool %s references machine config %s, which does not declare a version", pool.GetName()) 
```

Comment (wking) [reply pkg/operator/status.go]: Oops.  Fixed with 38beced -> 4ab79a5.

**Pull request merged**